pydantic/_internal/_generate_schema.py

```diff
@@ -1471,7 +1471,11 @@ def _dataclass_schema(
                 dataclass = origin
 
             config = getattr(dataclass, '__pydantic_config__', None)
-            with self._config_wrapper_stack.push(config), self._types_namespace_stack.push(dataclass):
+            with self._config_wrapper_stack.push(config), ExitStack() as dataclass_bases_stack:
+                for dataclass_base in dataclass.__mro__:
+                    if dataclasses.is_dataclass(dataclass_base):
+                        dataclass_bases_stack.enter_context(self._types_namespace_stack.push(dataclass_base))
```

Comment:
```suggestion
            with ExitStack() as dataclass_bases_stack:
                for dataclass_base in dataclass.__mro__:
                    if dataclasses.is_dataclass(dataclass_base):
                        config = getattr(dataclass_base, '__pydantic_config__', None)
                        dataclass_bases_stack.enter_context(self._config_wrapper_stack.push(config))
                        dataclass_bases_stack.enter_context(self._types_namespace_stack.push(dataclass_base))
```
Do we need to do something like this to handle configs in vanilla dataclasses (via setting the `__pydantic_config__` attribute on the class, even if it's not a pydantic dataclass)?

Reply:
Ah, yes, good point. In this case I think the MRO order does need to be reversed, since the config wrapper's tail is just the last config, not a merge of configs. I'll add this and some tests tomorrow.

Reply:
Code (and tests) added, and it did need to be reversed which is a bit inelegant, but oh well. I added a comment as well to explain why we need two loops.